### PR TITLE
feat: get friend requests

### LINKED ISSUE #697: Get friend request list and add friend requests
- [x] Update user state to add message bodies along with id and timestamp
- [ ] Add tests for RPC flow

### PR DESCRIPTION
**Pull request by Julieta11**
# What? 
Base branch for the New Friend Request feature. Find more info about the proposal [here](https://www.notion.so/decentraland/PRD-New-Friend-Requests-188555225272448f91f3eea5e84f1cd3) and check the RFC [here](https://adr.decentraland.org/adr/ADR-137). **The messages exchange between kernel and renderer is defined by RPC calls.**

So far, the PR addresses:
- Get pending requests - Closes #697

### CODE REVIEW

File: packages/shared/friends/sagas.ts [hunk 1]
```diff
@@ -708,6 +715,78 @@ export async function getFriendRequests(request: GetFriendRequestsPayload) {
   getUnityInstance().AddFriendRequests(addFriendRequestsPayload)
 }
 
+// New friend request flow
+export async function getFriendRequestsNew(request: GetFriendRequestsPayload) {
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:719]: Let's rename this to `getFriendRequestsProtocol` or `getFriendRequestsV2` FTTB, as new will shortly not be so.

File: packages/shared/friends/sagas.ts [hunk 2]
```diff
@@ -708,6 +715,78 @@ export async function getFriendRequests(request: GetFriendRequestsPayload) {
...
+  try {
+    // Get friends
+    const friends: FriendsState = getPrivateMessaging(store.getState())
+
+    const realmAdapter = await ensureRealmAdapterPromise()
+    const fetchContentServerWithPrefix = getFetchContentUrlPrefixFromRealmAdapter(realmAdapter)
+
+    //
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:727]: Remove empty comment.

File: packages/shared/friends/sagas.ts [hunk 3]
```diff
@@ -708,6 +715,78 @@ export async function getFriendRequests(request: GetFriendRequestsPayload) {
...
+      requestedTo: toFriendRequests.map((friend) => getFriendRequestInfo(friend, false)),
+      requestedFrom: fromFriendRequests.map((friend) => getFriendRequestInfo(friend, true)),
+      totalReceivedFriendRequests: friends.fromFriendRequests.length,
+      totalSentFriendRequests: friends.toFriendRequests.length
+    }
+
+    // Return requests
+    return friendRequests
+  } catch {
+    throw new Error()
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:760]: Don't catch a specific error and rethrow it as a generic one.

File: packages/shared/friends/utils.ts [hunk 4]
```diff
@@ -70,3 +70,39 @@ export function getMaxChannels(store: RootMetaState): number {
...
+ * The rule is: `ownId` < `otherUserId` ? `ownId_otherUserId` : `otherUserId_ownId`
+ * @param friendRequestId
+ * @return `{ ownId, otherUserId }`
+ */
+export function decodeFriendRequestId(friendRequestId: string) {
+  const firstUserId = friendRequestId.split('_')[0]
+  const secondUserId = friendRequestId.split('_')[1]
+
+  return firstUserId < secondUserId
+    ? { ownId: firstUserId, otherUserId: secondUserId }
```
Comment (moliva) [inline packages/shared/friends/utils.ts:106]: How are we sure `ownId` is the current user and `otherUserId` is the other one?

File: packages/shared/friends/sagas.ts [hunk 5]
```diff
@@ -708,6 +719,84 @@ export async function getFriendRequests(request: GetFriendRequestsPayload) {
...
+    const friends: FriendsState = getPrivateMessaging(store.getState())
+
+    const realmAdapter = await ensureRealmAdapterPromise()
+    const fetchContentServerWithPrefix = getFetchContentUrlPrefixFromRealmAdapter(realmAdapter)
+
+    const fromFriendRequests = friends.fromFriendRequests.slice(
+      request.receivedSkip,
+      request.receivedSkip + request.receivedLimit
+    )
+    const toFriendRequests = friends.toFriendRequests.slice(request.sentSkip, request.sentSkip + request.sentLimit)
```
Comment (kuruk-mm) [inline packages/shared/friends/sagas.ts:735]: Can you add comments explaining what these vector operations are doing? Looks like it is to scroll through the requests

**Review by kuruk-mm — APPROVED** [review]
Good job! It looks great! 👏

**Review by moliva — APPROVED** [review]
LGTM!

**Pull request merged**